commit 2167d08bde49ea9504cbf9214c913fcb83c7968e
Author: Alexander K <ak@natsys-lab.com>
Date:   Tue Aug 21 00:23:49 2018 +0300

    Make irq_fpu_usable() return true for Tempesta softirqs so that
    SIMD crypto algorithms won't be called through cryptd.

diff --git a/arch/x86/kernel/fpu/core.c b/arch/x86/kernel/fpu/core.c
index 532480a4..d27c08f4 100644
--- a/arch/x86/kernel/fpu/core.c
+++ b/arch/x86/kernel/fpu/core.c
@@ -86,6 +86,10 @@ static bool interrupted_user_mode(void)
  */
 bool irq_fpu_usable(void)
 {
+#ifdef CONFIG_SECURITY_TEMPESTA
+	if (likely(in_serving_softirq()))
+		return true;
+#endif
 	return !in_interrupt() ||
 		interrupted_user_mode() ||
 		interrupted_kernel_fpu_idle();
